tests/e2e/pom/agileway/basePage.page.ts

```diff
@@ -21,8 +21,7 @@ export class Basepage{
     await expect(this.alert).toBeVisible();
     await expect(this.alert).toContainText(alertMessage);
   }
-  
-  async verifyURLContains(Url: string){
-    await expect(this.page).toHaveURL(Url);
+  async verifyHeaderContains(headerName: string){
+    await expect(this.header.getByText(headerName)).toBeVisible();
```

Comment:
Why with headers you are looking for an element with text, when with alert - you first verify elements visibility and then text?

Which option do you think is better and why?